### PR TITLE
feat: adds support for app suite links dropdown menu

### PR DESCRIPTION
**Pull request by Golodhros**
### Summary of Changes
Adds support for a configurable app suite links dropdown
Adds docs and tests
Reorganizes application docs, adds placeholders to missing items
Reworks styling and border radius of all popovers and modals

### Screenshots
ams-app-suite

### Tests
Updated tests

### Documentation
Added docs
Added story
app-suite-story

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

File: frontend/docs/application_config.md [hunk 1]
```diff
@@ -245,6 +392,81 @@ navTheme: 'light',
 Which would render like the following:
 <img src='img/header-light-default-logo.png' width='50%' />
 
+## Nested Columns
+*TODO
```
Comment (Golodhros) [inline frontend/docs/application_config.md:396]: Would love your help @kristenarmes with this one

File: frontend/docs/application_config.md [hunk 2]
```diff
@@ -502,83 +725,17 @@ Example of this option enabled on tables and dashboards:
   },
 ```
 
+## Search Pagination
+*TODO
 
 ## Table Lineage
-
-_TODO: Please add doc_
+*TODO
```
Comment (Golodhros) [inline frontend/docs/application_config.md:732]: Would love your help @allisonsuarez with this one

File: frontend/amundsen_application/static/js/components/Tour/styles.scss [hunk 3]
```diff
@@ -1,7 +1,13 @@
 // Copyright Contributors to the Amundsen project.
 // SPDX-License-Identifier: Apache-2.0
 
+@import 'variables';
+
 /* Tour step tooltip content container */
+.react-joyride__tooltip {
+  border-radius: $popover-border-radius !important;
```
Comment (allisonsuarez) [inline frontend/amundsen_application/static/js/components/Tour/styles.scss:8]: is it necessary to use `!important` here?

Comment (Golodhros) [reply frontend/amundsen_application/static/js/components/Tour/styles.scss:8]: Yeah, not ideal but yes.
It is usually OK to do this when dealing with third party component customization though.

File: frontend/docs/application_config.md [hunk 4]
```diff
@@ -211,6 +249,62 @@ Introducing dashboards into Amundsen allows users to discovery data analysis tha
 
 After ingesting dashboard metadata into the search and metadata services, set `IndexDashboardsConfig.enabled` to `true` on the application configuration to display the UI for the aforementioned features.
 
+### Index Features
+*TODO
```
Comment (allisonsuarez) [inline frontend/docs/application_config.md:253]: is there something we need to add here?

File: frontend/docs/application_config.md (hunk 2 above)
Comment (allisonsuarez) [reply frontend/docs/application_config.md:732]: does this include the Amundsen graph UI or just the up/downstream lineage tabs?

File: frontend/amundsen_application/static/js/features/NavBar/index.tsx [hunk 5]
```diff
@@ -245,6 +299,24 @@ export const NavBar: React.FC<NavBarProps> = ({ loggedInUser, location }) => {
     setRunTour(false);
   }, [pathname]);
 
+  const handleAppSuiteToggle = (isOpen: boolean) => {
+    logAction({
+      target_id: '',
+      command: 'click',
+      target_type: 'button',
+      label: isOpen ? 'Open App Suite Menu' : 'Close App Suite Menu',
```
Comment (allisonsuarez) [inline frontend/amundsen_application/static/js/features/NavBar/index.tsx:307]: can we make these strings into constants?

File: frontend/docs/application_config.md (hunk 4 above)
Comment (Golodhros) [reply frontend/docs/application_config.md:253]: Yes, there is info for the other index options, but not for this one.
Ideally a short explanation and an example of how to set the flag (trivial, I know)

File: frontend/amundsen_application/static/js/features/NavBar/index.tsx (hunk 5 above)
Comment (Golodhros) [reply frontend/amundsen_application/static/js/features/NavBar/index.tsx:307]: I usually don't do it for tracking messages, but yes, I can do that.

File: frontend/docs/application_config.md (hunk 2 above)
Comment (Golodhros) [reply frontend/docs/application_config.md:732]: I leave it up to you!

File: frontend/docs/application_config.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Golodhros) [reply frontend/docs/application_config.md]: Thanks a lot!

**Review by kristenarmes — APPROVED** [review]

**Pull request merged**